### DISCUSSION TITLE
How to install dependencies

### DISCUSSION THREAD

**Discussion by cLonata** [adonisjs/core#3172] [Help] [answered]
Hello,
I am trying to switch over to AdonisJS but I haven't seen any on the docs about adding dependencies on to the slim package that I installed. What I have tried:

`$ node ace invoke @adonisjs/shield

Error: Cannot invoke instructions. Missing package "@adonisjs/shield"`

Is it wrong procedure or should I just be using npm to install those dependencies?

**Comment by RomainLanz** [comment, marked as answer]
Hey @cLonata! 👋 

You should first install those dependencies.
Once installed, you can use the `node ace configure <package name>` command to run the scaffold (if there is any) of those packages.

Ensure to check the installation steps we have [in our documentation](https://docs.adonisjs.com/guides/security/web-security).

**Reply by cLonata** [reply]
Thanks for answering!
Oh my bad I guess I missed that one! On the other side, do you guys also have a list of dependencies that are available to use?
Also will there be any problem if I don't use the built in validation? I am currently are used to using hapi and joi for the validation part and I think it would took me way too long to re learn everything over.
